hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -450,17 +435,32 @@ private void writeTableMetadata(HoodieTable<T, JavaRDD<HoodieRecord<T>>, JavaRDD
       }
       metaClient.reloadActiveTimeline();
 
-      // re-bootstrap metadata table if required
       initializeMetadataTable(Option.of(instantTime));
     }
     metaClient.validateTableProperties(config.getProps(), operationType);
     return getTableAndInitCtx(metaClient, operationType, instantTime);
   }
 
+  /**
+   * Initialize the metadata table if needed. Creating the metadata table writer
+   * will trigger the initial bootstrapping from the data table.
+   * <p>
+   * TODO: HUDI-2862 Guard the initial bootstrapping with transaction lock so
+   * as to make the initial metadata table bootstrapping single threaded.
+   *
+   * @param inFlightInstantTimestamp - The in-flight action responsible for the metadata table initialization
+   */
+  private void initializeMetadataTable(Option<String> inFlightInstantTimestamp) {
```

Comment:
where is this being called? I thought we have removed all callers?

Reply:
This is called from the `getTableAndInitCtx()` just after the table upgrade.